include/cloysterhpc/dbus_client.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <any>
+#include <memory>
+#include <sdbus-c++/sdbus-c++.h>
+#include <string>
+#include <type_traits>
+#include <variant>
+
+#include <cloysterhpc/messagebus.h>
+
+class DBusClient : public MessageBus {
+
+private:
+    std::unique_ptr<sdbus::IProxy> m_proxy;
+
+    //    std::unique_ptr<sdbus::IProxy>& getProxy();
+    sdbus::MethodCall createMethodCall(
+        std::string interface, std::string method);
+
+public:
+    DBusClient(std::string bus, std::string object)
+        : m_proxy(
+            sdbus::createProxy(std::move(sdbus::createSystemBusConnection()),
+                std::move(sdbus::ServiceName { bus }),
+                std::move(sdbus::ObjectPath { object })))
+    {
+    }
+
+    sdbus::MethodReply callMethod(const sdbus::MethodCall& message);
+
+    std::unique_ptr<MessageBusMethod> method(
+        std::string interface, std::string method);
+
+    virtual ~DBusClient() { }
+};
+
+class DBusMethod : public MessageBusMethod {
+    friend class DBusClient;
+
+private:
+    DBusClient* m_client;
```

Comment:
We do not own this pointer.
Usually, we pass the `this` pointer of the DBusClient here, check the concrete implementation

```cpp
    std::unique_ptr<MessageBusMethod> mtd(
        new DBusMethod { this, std::move(m) });
    return mtd;
```

For some reason also, references did not work here
And a `shared_ptr` would delete the pointer, since we would have to create it here, or force everyone to use a shared_ptr, if we go to the route of using [enable_shared_from_this](https://en.cppreference.com/w/cpp/memory/enable_shared_from_this)